shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -560,50 +560,57 @@ v8::MaybeLocal<v8::Object> CreateProxyForAPI(
   }
 }
 
-void ExposeAPIInMainWorld(v8::Isolate* isolate,
-                          const std::string& key,
-                          v8::Local<v8::Value> api,
-                          gin_helper::Arguments* args) {
-  TRACE_EVENT1("electron", "ContextBridge::ExposeAPIInMainWorld", "key", key);
+void ExposeAPIInWorld(v8::Isolate* isolate,
+                      const int world_id,
+                      const std::string& key,
+                      v8::Local<v8::Value> api,
+                      gin_helper::Arguments* args) {
+  TRACE_EVENT2("electron", "ContextBridge::ExposeAPIInWorld", "key", key,
+               "worldId", world_id);
 
   auto* render_frame = GetRenderFrame(isolate->GetCurrentContext()->Global());
   CHECK(render_frame);
   auto* frame = render_frame->GetWebFrame();
   CHECK(frame);
+
   v8::Local<v8::Context> main_context = frame->MainWorldScriptContext();
-  gin_helper::Dictionary global(main_context->GetIsolate(),
-                                main_context->Global());
+  v8::Local<v8::Context> context =
+      world_id == 0 ? main_context
+                    : frame->GetScriptContextFromWorldId(isolate, world_id);
+
+  gin_helper::Dictionary properties(context->GetIsolate(), context->Global());
 
-  if (global.Has(key)) {
+  if (properties.Has(key)) {
     args->ThrowError(
         "Cannot bind an API on top of an existing property on the window "
         "object");
     return;
   }
 
-  v8::Local<v8::Context> isolated_context = frame->GetScriptContextFromWorldId(
-      args->isolate(), WorldIDs::ISOLATED_WORLD_ID);
+  v8::Local<v8::Context> electron_isolated_context =
+      frame->GetScriptContextFromWorldId(args->isolate(),
+                                         WorldIDs::ISOLATED_WORLD_ID);
 
   {
     context_bridge::ObjectCache object_cache;
     v8::Context::Scope main_context_scope(main_context);
```

Comment:
The `v8::Context::Scope` needs be constructed using the context which will be modified. In my other comment, I used `target_context` to refer to it.